### PR TITLE
update _common.json format

### LINKED ISSUE #46744: The JSON spec parser should support the new format
While working on https://github.com/elastic/elasticsearch/pull/46522, @polyfractal has discovered that the [parsing for `_common.json`](https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/test/rest/yaml/restspec/ClientYamlSuiteRestSpec.java#L103-L123) assumes there is only one object to parse, while the new spec format there are two objects.

The parser needs to be updated to adopt the new structure of the JSON spec.
Following, the difference between the old format and the new one.

```diff
-  "description": "Parameters that are accepted by all API endpoints.",
-  "documentation": "https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html",
+  "documentation": {
+    "url": "https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html",
+    "description": "Parameters that are accepted by all API endpoints."
+  },
```

cc @Mpdreamz

### PR DESCRIPTION
**Pull request by javanna**
API spec now use an object for the documentation field. _common was not updated yet. This commit updates _common.json and its corresponding parser.

Closes #46744

### CODE REVIEW

**Comment by javanna** [comment]
cc @elastic/es-clients

**Review by delvedor — APPROVED** [review]
LGTM, this should also be backported to `7.x` :)

File: rest-api-spec/src/main/resources/rest-api-spec/api/_common.json [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
-  "description": "Parameters that are accepted by all API endpoints.",
-  "documentation": "https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html",
+  "documentation" : {
+    "url": "Parameters that are accepted by all API endpoints.",
+    "documentation": "https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html"
```
Comment (delvedor) [inline rest-api-spec/src/main/resources/rest-api-spec/api/_common.json]:
```suggestion
    "url": "https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html"
```

File: rest-api-spec/src/main/resources/rest-api-spec/api/_common.json [hunk 2]
```diff
@@ -1,6 +1,8 @@
 {
-  "description": "Parameters that are accepted by all API endpoints.",
-  "documentation": "https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html",
+  "documentation" : {
+    "url": "Parameters that are accepted by all API endpoints.",
```
Comment (delvedor) [inline rest-api-spec/src/main/resources/rest-api-spec/api/_common.json]:
```suggestion
    "description": "Parameters that are accepted by all API endpoints.",
```

File: rest-api-spec/src/main/resources/rest-api-spec/api/_common.json (hunk 1 above)
Comment (javanna) [reply rest-api-spec/src/main/resources/rest-api-spec/api/_common.json]: haha

**Pull request merged**